### PR TITLE
[TACHYON-1674] Improve Running-Tachyon-on-EC2 doc.

### PR DESCRIPTION
**Pull request by ifcharming**
https://tachyon.atlassian.net/browse/TACHYON-1674

1. Link to EC2 tutorial.
2. Highlight zone should be changed if necessary.
3. Add spot instance section.
4. Minor improvement for console monitoring.

### CODE REVIEW

**Comment by aaudiber** [comment]
Thanks for improving the docs @ifcharming! Do you mind posting a screenshot of what the new docs look like so we can see the formatting?

File: docs/Running-Tachyon-on-EC2.md [hunk 1]
```diff
@@ -113,7 +127,14 @@ Default port for Hadoop Web UI is **50070**.
 Visit `http://{MASTER_IP}:{PORT}` in the browser to access the Web UIs.
 
 You can also monitor the instances state through
-[AWS web console](https://console.aws.amazon.com/console/home?region=us-east-1).
+[AWS web console](https://console.aws.amazon.com/console).
+Make sure you are in the console for the region you started the cluster.
+
+Here are some scenarios when you may want to check the console:
+ - when ./create failed
```
Comment (apc999) [inline docs/Running-Tachyon-on-EC2.md]: <pre>`./create`</pre>?

File: docs/Running-Tachyon-on-EC2.md [hunk 2]
```diff
@@ -113,7 +127,14 @@ Default port for Hadoop Web UI is **50070**.
...
+ - after you call ./destroy
```
Comment (apc999) [inline docs/Running-Tachyon-on-EC2.md]: ditto

**Comment by ifcharming** [comment]
@aaudiber , here is a screenshot: 
screen shot 2016-01-27 at 3 53 44 pm

But I guess you can also view the markdown preview by clicking
"View" button in "Files changed" section?

**Comment by ifcharming** [comment]
Wording updated. PTAL.

**Comment by apc999** [comment]
+1 LGTM, thanks for the improvement

**Comment by apc999** [comment]
@ifcharming could you resolve conflicts on this PR?

**Comment by ifcharming** [comment]
Conflicts resolved.

To preview final version with format: https://github.com/ifcharming/tachyon/blob/fix-ec2-doc/docs/en/Running-Tachyon-on-EC2.md

File: docs/en/Running-Tachyon-on-EC2.md [hunk 3]
```diff
@@ -55,6 +55,8 @@ $ sudo pip install -r pip-req.txt
 To run a Tachyon cluster on EC2, first sign up for an Amazon EC2 account
 on the [Amazon Web Services site](http://aws.amazon.com/).
 
+If you are not familiar with Aamazon EC2, you can read [this tutorial](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html) first.
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: Aamazon typo

File: docs/en/Running-Tachyon-on-EC2.md [hunk 4]
```diff
@@ -85,7 +87,20 @@ By default, the Vagrant script creates a
 named *tachyon-vagrant-test* at
 [Region(**us-east-1**) and Availability Zone(**us-east-1a**)](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html).
 The security group will be set up automatically in the region with all inbound/outbound network
-traffic opened. You can change the *security group*, *region* and *availability zone* in `ec2.yml`.
+traffic opened. You can change the *security group*, *region* and *availability zone* in `ec2.yml`. Sometimes the default zone can be unavailable.
+Note: the keypair is associated with a sepcific zone. If you ran into permission/connection errors, please first check the zone.
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: sepcific typo

maybe give an example, e.g. US East

File: docs/en/Running-Tachyon-on-EC2.md [hunk 5]
```diff
@@ -85,7 +87,20 @@ By default, the Vagrant script creates a
...
+
+**Spot instances**
+
+Using spot instance is a way to reduce EC2 cost. Spot instances are non-guaranteed instances which are priced with bidding.
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: instance -> instances

File: docs/en/Running-Tachyon-on-EC2.md [hunk 6]
```diff
@@ -113,7 +128,14 @@ Default port for Hadoop Web UI is **50070**.
 Visit `http://{MASTER_IP}:{PORT}` in the browser to access the Web UIs.
 
 You can also monitor the instances state through
-[AWS web console](https://console.aws.amazon.com/console/home?region=us-east-1).
+[AWS web console](https://console.aws.amazon.com/console).
+Make sure you are in the console for the region you started the cluster.
+
+Here are some scenarios when you may want to check the console:
+ - when failed to create the cluster
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: when cluster creation fails

File: docs/en/Running-Tachyon-on-EC2.md [hunk 7]
```diff
@@ -113,7 +128,14 @@ Default port for Hadoop Web UI is **50070**.
...
+ - after destroying the cluster
+ - when you no longer need the cluster
+
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: extra newline

**Comment by aaudiber** [comment]
@ifcharming It's good to know that we can use github as a markdown viewer! I've left a few minor comments in the code

File: docs/en/Running-Tachyon-on-EC2.md [hunk 8]
```diff
@@ -85,7 +87,20 @@ By default, the Vagrant script creates a
...
+**Spot instances**
+
+Using spot instance is a way to reduce EC2 cost. Spot instances are non-guaranteed instances which are priced with bidding.
+Note that spot instances may be taken away from you if someone bids more, and there are no more spot instances available.
+However, for short-term testing, spot instances are very appropriate, because it is rare that spot instances are taken from you.
+
+By default, the deploy scripts DO NOT use spot instances. Therefore, you have to enable deploy scripts to use spot instances.
+
+In order to enable spot instances, you have to modify the file: deploy/vagrant/conf/ec2.yml:
```
Comment (apc999) [inline docs/en/Running-Tachyon-on-EC2.md]: \`deploy/vagrant/conf/ec2.yml\` ?

File: docs/en/Running-Tachyon-on-EC2.md [hunk 9]
```diff
@@ -113,7 +128,13 @@ Default port for Hadoop Web UI is **50070**.
 Visit `http://{MASTER_IP}:{PORT}` in the browser to access the Web UIs.
 
 You can also monitor the instances state through
-[AWS web console](https://console.aws.amazon.com/console/home?region=us-east-1).
+[AWS web console](https://console.aws.amazon.com/console).
+Make sure you are in the console for the region you started the cluster.
```
Comment (apc999) [inline docs/en/Running-Tachyon-on-EC2.md]: Make sure that ... the region where you started the cluster

File: docs/en/Running-Tachyon-on-EC2.md [hunk 10]
```diff
@@ -113,7 +128,13 @@ Default port for Hadoop Web UI is **50070**.
...
+
+Here are some scenarios when you may want to check the console:
```
Comment (apc999) [inline docs/en/Running-Tachyon-on-EC2.md]: could we be a bit more specific here, e.g., check the console for what?

**Comment by apc999** [comment]
Made another pass, left three minor comments.

**Comment by ifcharming** [comment]
Thanks @aaudiber and @apc999 , updated, PTAL.

File: docs/en/Running-Tachyon-on-EC2.md [hunk 11]
```diff
@@ -113,7 +128,13 @@ Default port for Hadoop Web UI is **50070**.
...
 
 You can also monitor the instances state through
-[AWS web console](https://console.aws.amazon.com/console/home?region=us-east-1).
+[AWS web console](https://console.aws.amazon.com/console).
+Make sure that you are in the console for the region where you started the cluster.
+
+Here are some scenarios when you may want to check the console:
+ - When the cluster creation fails, check EC2 instances status/logs.
+ - After the cluster is destroyed, confirm EC2 instances are terminated.
+ - When you no longer need the cluster, make sure EC2 instances and S3 buckets are NOT costing you extra money.
```
Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: this document isn't about S3 - we can remove "and S3"

Comment (ifcharming) [inline docs/en/Running-Tachyon-on-EC2.md]: But as the default UNDERFS with EC2 has been changed to S3, would it make sense to just double check S3 bucket deleted as expected?

Comment (aaudiber) [inline docs/en/Running-Tachyon-on-EC2.md]: The S3 bucket isn't part of the cluster - it should exist independently. The create script doesn't create the bucket - it expects it to already exist.

Comment (ifcharming) [inline docs/en/Running-Tachyon-on-EC2.md]: I see, updated, thanks!

**Comment by ifcharming** [comment]
ok to test

**Comment by aaudiber** [comment]
LGTM

**Comment by yupeng9** [comment]
btw, the EC2 doc is outdated, since now the default storage is changed to s3.

**Comment by ifcharming** [comment]
Can we make this PR just for the EC2 related improvements, using Hadoop for now?
I'll add a note in the doc, clarifying that default underfs for EC2 is s3. But it's showing the example using hadoop2.

**Comment by ifcharming** [comment]
Updated, PTAL.

**Comment by uronce-cc** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM. Thanks.

**Pull request merged**